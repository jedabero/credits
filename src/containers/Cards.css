.cardrow {
  display: grid;
  grid-template-columns: repeat(6, 1fr) 2fr;
}

.cardrow > input {
  display: block;
  min-width: 0;
}
