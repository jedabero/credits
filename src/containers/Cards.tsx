import { useCallback, useState } from 'react';
import { EntityId } from '@reduxjs/toolkit';
import CardForm from '../components/CardForm';
import CardsTable from '../components/CardsTable';

import './Cards.css';

type CardsProps = {
  onViewCardMovements: React.Dispatch<React.SetStateAction<EntityId | undefined>>,
};

function Cards({ onViewCardMovements }: CardsProps) {
  const [cardIdBeingEdited, setCardId] = useState<EntityId | undefined>();
  const clearCardId = useCallback(() => {
    setCardId(undefined);
  }, [setCardId]);
  return (
    <div className="cards">
      <h3>Cards</h3>
      <CardForm cardId={cardIdBeingEdited} clearCardId={clearCardId} />
      <CardsTable onEdit={setCardId} onView={onViewCardMovements} />
    </div>
  );
}

export default Cards;
