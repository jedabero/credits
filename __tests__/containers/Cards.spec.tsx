/* eslint-disable import/no-extraneous-dependencies */
import React, { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import Cards from '../../src/containers/Cards';

describe('Cards test', () => {
  const renderer = createRenderer();
  let output: ReactElement;
  test('render Cards', () => {
    renderer.render(<Cards onViewCardMovements={() => {}} />);
    output = renderer.getRenderOutput<ReactElement>();
    expect(output).toMatchSnapshot();
  });
  test('set cardId', () => {
    output.props.children[2].props.onEdit('SOME ID');
    output = renderer.getRenderOutput<ReactElement>();
    expect(output.props.children[1].props.cardId).toBe('SOME ID');
    expect(output).toMatchSnapshot();
  });
  test('call clearCardId', () => {
    output.props.children[1].props.clearCardId();
    output = renderer.getRenderOutput<ReactElement>();
    expect(output.props.children[1].props.cardId).toBeNull();
    expect(output).toMatchSnapshot();
  });
});
