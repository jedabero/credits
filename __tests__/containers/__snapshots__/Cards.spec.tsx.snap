// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards test call clearCardId 1`] = `
<div
  className="cards"
>
  <h3>
    Cards
  </h3>
  <CardForm
    cardId={null}
    clearCardId={[Function]}
  />
  <CardsTable
    onEdit={[Function]}
    onView={[Function]}
  />
</div>
`;

exports[`Cards test render Cards 1`] = `
<div
  className="cards"
>
  <h3>
    Cards
  </h3>
  <CardForm
    cardId={null}
    clearCardId={[Function]}
  />
  <CardsTable
    onEdit={[Function]}
    onView={[Function]}
  />
</div>
`;

exports[`Cards test set cardId 1`] = `
<div
  className="cards"
>
  <h3>
    Cards
  </h3>
  <CardForm
    cardId="SOME ID"
    clearCardId={[Function]}
  />
  <CardsTable
    onEdit={[Function]}
    onView={[Function]}
  />
</div>
`;
